src/plugins/backgrounds/unsplash/Unsplash.tsx

```diff
@@ -14,23 +13,33 @@ const Unsplash: FC<Props> = ({
   setCache,
 }) => {
   const cacheArea = { cache, setCache };
-  const image = useRotatingCache(
-    () => getImage(data, loader),
+
+  useRotatingCache(
+    () =>
+      getImage(data, loader).then(imageData => {
+        var prevImages = Array<Image>();
+        if (cache) var prevImages = cache.next.previous_images;
+        if (prevImages.length > 2) prevImages.shift();
+        return {
+          currentImage: imageData,
+          //cache the last 3 images
+          previous_images: [...prevImages!, imageData],
+        };
+      }),
     cacheArea,
     data.timeout * 1000,
     [data.by, data.collections, data.featured, data.search],
   );
 
-  const url = useObjectUrl(image && image.data);
-
+  const url = cache && URL.createObjectURL(cache.now.currentImage.data);
```

Comment:
I ran into a few errors with this line. If you run into a type error with this line, you will have to clear site data or open it in an unused profile. Maybe there should be some form of fallback